### PR TITLE
[WFLY-13351] Galleon layer for the Mail subsystem

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFLY-13351

### CODE REVIEW

**Comment by darranl** [comment]
For all new layers I think we need to make sure security is considered in the requirements,

We don't want to accidentally leave anything unsecured by default but ideally if layers can avoid a dependency on any security layers that would be preferred.

Failing that for the new layers if security dependencies are required we should be preferring elytron with an option to select legacy security.  At some point in the future all legacy security related layers will be removed.

Can we please add a check for these questions in the requirements?

**Review by fabiobrz — CHANGES_REQUESTED** [review]
Hi @yersan - thanks for this proposal. I dropped just one comment, let me know what you think.

File: mail/WFLY-13351_galleon_layer_for_mail_subsystem.adoc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+=== Affected Projects or Components
+
+Wildfly - Mail subsystem
+
+== Requirements
+
+=== Hard Requirements
+
+* The Galleon layer must provide the basic configuration Mail subsystem configuration.
```
[the remark this answers is not in the document]
Comment (yersan) [reply mail/WFLY-13351_galleon_layer_for_mail_subsystem.adoc]: @fabiobrz I've just fixed the wording here.

**Comment by yersan** [comment]
> For all new layers I think we need to make sure security is considered in the requirements,
> 
> We don't want to accidentally leave anything unsecured by default but ideally if layers can avoid a dependency on any security layers that would be preferred.
> 
> Failing that for the new layers if security dependencies are required we should be preferring elytron with an option to select legacy security. At some point in the future all legacy security related layers will be removed.
> 
> Can we please add a check for these questions in the requirements?

I've just added a note in the requirements section specifying this Galleon layer will not depend upon a security layer

**Review by fabiobrz — APPROVED** [review]

**Pull request merged**